packages/core/src/lib/server/createAdminUIMiddleware.ts

```diff
@@ -10,28 +10,33 @@ const adminErrorHTMLFilepath = path.join(
   'admin-error.html'
 );
 
-export const createAdminUIMiddleware = async (
-  config: KeystoneConfig,
-  createContext: CreateContext,
-  dev: boolean,
-  projectAdminPath: string
-) => {
+type NextApp = {
+  prepare(): Promise<void>;
+  getRequestHandler(): express.Application;
+  render(req: express.Request, res: express.Response, url: string): void;
+};
+
+export async function getNextApp(dev: boolean, projectAdminPath: string): Promise<NextApp> {
   /** We do this to stop webpack from bundling next inside of next */
-  const { ui, graphql, session } = config;
   const _next = 'next';
   const next = require(_next);
-  const app = next({ dev, dir: projectAdminPath });
-  const handle = app.getRequestHandler();
+  const app = next({ dev, dir: projectAdminPath }) as NextApp;
   await app.prepare();
+  return app;
+}
 
+export function createAdminUIMiddlewareWithNextApp(
+  config: KeystoneConfig,
+  createContext: CreateContext,
+  nextApp: NextApp
+) {
+  const handle = nextApp.getRequestHandler();
+
+  const { ui, session } = config;
   const publicPages = ui?.publicPages ?? [];
   return async (req: express.Request, res: express.Response) => {
     const { pathname } = url.parse(req.url);
-    if (
-      pathname?.startsWith('/_next') ||
-      pathname === (graphql?.path || '/api/graphql') ||
-      pathname === '/api/__keystone_api_build'
```

Comment:
These deletions don't fix anything, they just became unnecessary after #7910 and #7809 respectively